File: app/src/main/java/rango/tool/androidtool/base/list/adapter/BaseItemView.java
package rango.tool.androidtool.base.list.adapter;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

import java.lang.ref.WeakReference;

public abstract class BaseItemView extends RelativeLayout implements ViewWrapper.Binder<BaseItemData> {

    protected BaseItemData mData;
    protected WeakReference<Context> mContextWeakReference;

    public BaseItemView(Context context) {
        this(context, null);
    }

    public BaseItemView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public BaseItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        mContextWeakReference = new WeakReference<>(context);
    }

    @Override
    public void bind(BaseItemData data) {
        if (data != null && data.getData() != null) {
            mData = data;
            refreshUI();
        }
    }

    protected abstract void refreshUI();
}

File: wanandroid/src/main/java/rango/kotlin/wanandroid/common/list/BaseItemView.kt
package rango.kotlin.wanandroid.common.list

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import androidx.constraintlayout.widget.ConstraintLayout

abstract class BaseItemView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : ConstraintLayout(context, attrs, defStyleAttr), ViewWrapper.Binder {
    var mData: BaseItemData<*>? = null

    override fun bind(data: BaseItemData<*>?) {
        if (data?.data != null) {
            mData = data
            refreshUI()
        }
    }

    protected abstract fun refreshUI()
}